test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_neuronx.py

```
@@ -0,0 +1,150 @@
+    :param sagemaker_regions: List of SageMaker regions
+    :param helper_function: Function to invoke
+    :param helper_function_args: Helper function args
+
+    :return: None
+    """
+    from ..... import get_ecr_image_region, get_sagemaker_session, get_ecr_image
+
+    ecr_image_region = get_ecr_image_region(ecr_image)
+    for region in sagemaker_regions:
+        sagemaker_session = get_sagemaker_session(region)
+        # Reupload the image to test region if needed
+        tested_ecr_image = (
+            get_ecr_image(ecr_image, region) if region != ecr_image_region else ecr_image
+        )
+        try:
+            helper_function(tested_ecr_image, sagemaker_session, **helper_function_args)
+            return
+        except sagemaker.exceptions.UnexpectedStatusException as e:
+            if "CapacityError" in str(e):
+                raise ValueError("CapacityError: Retry.")
+            else:
+                raise e
+
+
+@pytest.mark.processor("neuronx")
+@pytest.mark.model("hf-pt-glue-neuronx")
+@pytest.mark.neuronx_test
+@pytest.mark.skip_py2_containers
+def test_neuronx_text_classification(ecr_image, sagemaker_regions, py_version, instance_type):
+    function_args = {
+        "py_version": py_version,
+        "instance_type": instance_type,
+        "instance_count": 1,
+        "num_neuron_cores": 2,
+    }
+    invoke_neuron_helper_function(
+        ecr_image, sagemaker_regions, _test_neuronx_text_classification_function, function_args
+    )
+
+
+def _test_neuronx_text_classification_function(
+    ecr_image,
+    sagemaker_session,
+    py_version,
+    instance_type="ml.trn1.2xlarge",
+    instance_count=1,
+    num_neuron_cores=2,
+):
+    pytorch_version = get_pytorch_version(ecr_image)
+    if pytorch_version in SpecifierSet("==1.13.*"):
+        optimum_neuron_version = "0.0.3"
+    else:
+        pytest.skip(f"`optimum_neuron_version` to be set for PyTorch version {pytorch_version}.")
```

Should we raise an error over here instead of skip to ensure that we do not end up skipping the test in the future?

---

Sounds good to me. Let's raise an error.